### DISCUSSION TITLE
Suggestion: Just tested swaybar's clipboard widget

### DISCUSSION THREAD

**Discussion by Zeioth** [Manjaro-Sway/manjaro-sway#257] [Ideas]
Changing the line onclick to this, re-copies the selected item into the active clipboard, which is probably the expected behavior.

    "on-click": "swaymsg -q exec 'cliphist list | wofi --show dmenu -p clipboard | cliphist decode | wl-copy'; pkill -RTMIN+9 waybar",

**Comment by Zeioth** [comment]
Also, consider adding a more informative tooltip like:
![screenshot_2022-02-07_05-32-21_284782569](https://user-images.githubusercontent.com/3357792/152724900-7e2514f7-33c3-43cb-9978-8507a753ac49.png)

    "exec": "printf '{\"tooltip\":\"%s\"}' $(cliphist list | wc -l)' item(s) in the clipboard\r(Mid click to clear)'",

**Reply by boredland** [reply]
Added your tooltip, but don't understand to other part.

**Comment by Mathew-D** [comment]
The tool tip change looks good but once you select an item from the list it is usage.  Wouldn't copying it back to the clipboard add another copy of it to the history?
And can you put it in PR for the tool tip change.

**Comment by Zeioth** [comment]
It doesn't add another item. It moves the selected item to the first position. Works as expected in practice.

Thank you for your review.
Sure if you tell me where I can access the actual OS code I wouldn't mind!

**Comment by Mathew-D** [comment]
To put in a PR you go to: https://github.com/Manjaro-Sway/desktop-settings
Then just go to community/sway/

**Comment by Zeioth** [comment]
I just tought maybe would be a good idea to flush the clipboard history on startup by default, to minimize the risk of storing passwords copy/pasted by the user for too long.

I'd even say one is very unlikely to look for content stored for more than 1h ever again.

**Reply by boredland** [reply]
It is being flushed already.

**Comment by boredland** [comment]
Ya I agree with a relatively short timespan to store this / delete on boot etc.

I kinda wanted this feature because I used sth. similar quite a lot in other environments. But it seems how I work changed and I barely used this clipboard feature since we introduced it. So: don't count my opinion. For me purging this on reboot sounds reasonable.
